index.ts

```diff
@@ -210,13 +236,16 @@ function getContentScriptMethod<
   TPublicMethod extends PublicMethodWithTarget<TMethod>
 >(type: TType, options: Options = {}): TPublicMethod {
   const publicMethod = (target: Target, ...args: Parameters<TMethod>) => {
-    const sendMessage = async () =>
-      browser.tabs.sendMessage(
-        target.tabId,
-        makeMessage(type, args),
-        // `frameId` must be specified. If missing, the message would be sent to every frame
-        { frameId: target.frameId ?? 0 }
-      );
+    // eslint-disable-next-line no-negated-condition -- Looks better
+    const sendMessage = !browser.tabs
+      ? async () => browser.runtime.sendMessage(makeMessage(type, args, target))
+      : async () =>
+          browser.tabs.sendMessage(
+            target.tabId,
+            makeMessage(type, args),
+            // `frameId` must be specified. If missing, the message is sent to every frame
+            { frameId: target.frameId ?? 0 }
+          );
```

Comment:
This is the current "routing" implementation. 

Since `getContentScriptMethod` is specific to content scripts, it's clear what the direction is: If we can't message the CS directly, we must contact the background page.